msi/release.sh

```diff
@@ -1,34 +1,119 @@
+    echo "       If this option is not specified, the installer will be written to your"
+    echo "       home directory."
+    echo
+    echo "   [Optional Components]"
+    echo "       A list of additional packages you want bundled with your Git for Windows"
+    echo "       distribution installer."
+    echo
+    echo "       For example, if you want to install emacs with your Git for Windows"
+    echo "       installation, you would run the command:"
+    echo
+    echo "           ./release.sh M.m.r.b mingw-w64-emacs-git"
+    echo
+    echo "       where M.m.r.b is the Major, minor, revision, and build number, "
+    echo "       respectively."
+    echo
+    exit 1
 }
 
 die () {
-	echo "$*" >&1
-	exit 1
+    echo "$*" >&2
+    exit 1
+}
+
+function parse_version() {
+    echo "$1" | grep -Px '\d+(\.(0|[1-9]+)){2,3}'
 }
 
 ARCH="$(uname -m)"
 case "$ARCH" in
 i686)
-	BITNESS=32
-	WIX_ARCH=x86
-	;;
+    BITNESS=32
+    WIX_ARCH=x86
+    ;;
 x86_64)
-	BITNESS=64
-	WIX_ARCH=x64
-	;;
+    BITNESS=64
+    WIX_ARCH=x64
+    ;;
 *)
-	die "Unhandled architecture: $ARCH"
-	;;
+    die "Unhandled architecture: $ARCH"
+    ;;
 esac
-VERSION=$1
-shift
-TARGET="$HOME"/Git-"$VERSION"-"$BITNESS"-bit.msi
+
+for arg
+do
+    case "$1" in
+        -o=*|--outputDir=*)
+            TARGET="${1##*=}"
+            shift
+        ;;
+        -o)
+            shift
+            TARGET="$1"
+            shift
+        ;;
+        -v=*|--version=*)
+            VERSION="$(parse_version $(echo "${1##*=}"))"
+            shift
+        ;;
+        -v)
+            shift
+            VERSION="$(parse_version "$1")"
+            shift
+        ;;
+        -*) die "Unrecognized option: $1" ;;
+        *)
+            if [ -z "$VERSION" ]; then
+                VERSION="$(parse_version $1)"
+                shift
+            fi
+            if [ "$#" -gt "0" ]; then
+                # We have to know if this is a "custom MSI build"
+                # That's because we'll need to update the MSI Product ID Guid in some way
+                # so that 1) we don't wipe out a user's custom git installation; and
+                # more importantly 2) a user doesn't "hijack" a standard git installation,
+                # either maliciously or unintentionally.
+                #
+                # This needs to be discussed further.
+                CUSTOM=1
+            fi
+            break
+        ;;
+    esac
+done
+if [ -z "$VERSION" ]; then
+    die "A version must be provided in the format <Major>.<Minor>.<Revision>.<Build/Patch>."
+fi
+TARGET="${TARGET-$HOME}"/Git-"$VERSION"-"$BITNESS"-bit.msi
+
+echo
+echo "Creating Git for Windows MSI Installer:"
+echo "---------------------------------------------------------------------"
+echo "    Output to: $TARGET"
+echo "    Version: $VERSION"
+echo "    Build Type: $((test -n "$CUSTOM" && test "$CUSTOM" -eq 1 && echo "Custom") || (test -n "$TEST" && test "$TEST" -eq "1" && echo "Test") || echo "Release")"
```

Comment:
It seemed to, but, you're right, it shouldn't have. Maybe it worked "by accident". In any event, I was wondering when I switched editors why syntax hilighting wasn't working correctly, and this line was the issue. Inserting a space fixed it. I will fix it accordingly.